Loose dependency in CompositeInjectingConstraintValidatorFactory

```
Lookup and initialization failed for a resource class: class org.glassfish.jersey.server.validation.internal.hibernate.HibernateInjectingConstraintValidatorFactory.
org.glassfish.hk2.api.MultiException: A MultiException has 1 exceptions.  They are:
1. java.lang.NoClassDefFoundError: org/glassfish/jersey/ext/cdi1x/internal/CdiUtil

  org.jvnet.hk2.internal.Utilities.justPostConstruct:894
  org.jvnet.hk2.internal.ServiceLocatorImpl.postConstruct:1026
  org.jvnet.hk2.internal.ServiceLocatorImpl.createAndInitialize:1074
  org.jvnet.hk2.internal.ServiceLocatorImpl.createAndInitialize:1064
  org.glassfish.jersey.inject.hk2.AbstractHk2InjectionManager.createAndInitialize:189
  org.glassfish.jersey.inject.hk2.ImmediateHk2InjectionManager.createAndInitialize:30
  org.glassfish.jersey.internal.inject.Injections.getOrCreate:106
  org.glassfish.jersey.server.JerseyResourceContextConfigurator.lambda$init$0:45
  org.glassfish.jersey.server.internal.JerseyResourceContext.getResource:83
  org.glassfish.jersey.server.validation.internal.CompositeInjectingConstraintValidatorFactory.postConstruct:45
  jdk.internal.reflect.NativeMethodAccessorImpl.invoke0:native
  jdk.internal.reflect.NativeMethodAccessorImpl.invoke:62
  jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke:43
  java.lang.reflect.Method.invoke:566
  org.glassfish.hk2.utilities.reflection.ReflectionHelper.invoke:1268
  org.jvnet.hk2.internal.Utilities.justPostConstruct:892
  org.jvnet.hk2.internal.ServiceLocatorImpl.postConstruct:1026
  org.jvnet.hk2.internal.ServiceLocatorImpl.createAndInitialize:1074
  org.jvnet.hk2.internal.ServiceLocatorImpl.createAndInitialize:1064
  org.glassfish.jersey.inject.hk2.AbstractHk2InjectionManager.createAndInitialize:189
  org.glassfish.jersey.inject.hk2.ImmediateHk2InjectionManager.createAndInitialize:30
  org.glassfish.jersey.internal.inject.Injections.getOrCreate:106
  org.glassfish.jersey.server.JerseyResourceContextConfigurator.lambda$init$0:45
  org.glassfish.jersey.server.internal.JerseyResourceContext.getResource:83
  org.glassfish.jersey.server.validation.internal.ValidationBinder$ConfiguredValidatorProvider.getDefaultValidatorContext:267
  org.glassfish.jersey.server.validation.internal.ValidationBinder$ConfiguredValidatorProvider.getDefaultValidator:245
  org.glassfish.jersey.server.validation.internal.ValidationBinder$ConfiguredValidatorProvider.get:187
  org.glassfish.jersey.server.validation.internal.ValidationBinder$ConfiguredValidatorProvider.get:157
  org.glassfish.jersey.inject.hk2.SupplierFactoryBridge.provide:76
  org.jvnet.hk2.internal.FactoryCreator.create:129
  org.jvnet.hk2.internal.SystemDescriptor.create:463
  org.jvnet.hk2.internal.PerLookupContext.findOrCreate:46
  org.jvnet.hk2.internal.Utilities.createService:2102
  org.jvnet.hk2.internal.ServiceLocatorImpl.internalGetService:758
  org.jvnet.hk2.internal.ServiceLocatorImpl.internalGetService:721
  org.jvnet.hk2.internal.ServiceLocatorImpl.getService:691
  org.glassfish.jersey.inject.hk2.AbstractHk2InjectionManager.getInstance:160
  org.glassfish.jersey.inject.hk2.ImmediateHk2InjectionManager.getInstance:30
  org.glassfish.jersey.server.model.internal.ResourceMethodInvokerConfigurator.lambda$postInit$0:54
  org.glassfish.jersey.server.model.ResourceMethodInvoker$Builder.build:194
  org.glassfish.jersey.server.internal.routing.RuntimeModelBuilder.createInflector:106
  org.glassfish.jersey.server.internal.routing.RuntimeModelBuilder.createMethodRouter:93
  org.glassfish.jersey.server.internal.routing.RuntimeModelBuilder.createResourceMethodRouters:287
  org.glassfish.jersey.server.internal.routing.RuntimeModelBuilder.buildModel:151
  org.glassfish.jersey.server.internal.routing.Routing$Builder.buildStage:223
  org.glassfish.jersey.server.ApplicationHandler.initialize:399
  org.glassfish.jersey.server.ApplicationHandler.lambda$initialize$1:293
  org.glassfish.jersey.internal.Errors.process:292
  org.glassfish.jersey.internal.Errors.process:274
  org.glassfish.jersey.internal.Errors.processWithException:232
  org.glassfish.jersey.server.ApplicationHandler.initialize:292
  org.glassfish.jersey.server.ApplicationHandler.<init>:259
  org.glassfish.jersey.servlet.WebComponent.<init>:311
  org.glassfish.jersey.servlet.ServletContainer.init:154
  org.glassfish.jersey.servlet.ServletContainer.init:347
  javax.servlet.GenericServlet.init:203
  org.eclipse.jetty.servlet.ServletHolder.initServlet:602
  org.eclipse.jetty.servlet.ServletHolder.initialize:413
  org.eclipse.jetty.servlet.ServletHandler.lambda$initialize$0:749
  java.util.stream.SortedOps$SizedRefSortingSink.end:357
  java.util.stream.AbstractPipeline.copyInto:485
  java.util.stream.AbstractPipeline.wrapAndCopyInto:474
  java.util.stream.StreamSpliterators$WrappingSpliterator.forEachRemaining:312
  java.util.stream.Streams$ConcatSpliterator.forEachRemaining:735
  java.util.stream.ReferencePipeline$Head.forEach:658
  org.eclipse.jetty.servlet.ServletHandler.initialize:773
  org.eclipse.jetty.servlet.ServletContextHandler.startContext:360
  org.eclipse.jetty.server.handler.ContextHandler.doStart:844
  org.eclipse.jetty.servlet.ServletContextHandler.doStart:275
  org.eclipse.jetty.util.component.AbstractLifeCycle.start:72
  org.eclipse.jetty.util.component.ContainerLifeCycle.start:169
  org.eclipse.jetty.util.component.ContainerLifeCycle.doStart:117
  org.eclipse.jetty.server.handler.AbstractHandler.doStart:97
  org.eclipse.jetty.util.component.AbstractLifeCycle.start:72
  org.eclipse.jetty.util.component.ContainerLifeCycle.start:169
  org.eclipse.jetty.util.component.ContainerLifeCycle.doStart:110
  org.eclipse.jetty.server.handler.AbstractHandler.doStart:97
  org.eclipse.jetty.server.handler.gzip.GzipHandler.doStart:425
  org.eclipse.jetty.util.component.AbstractLifeCycle.start:72
  org.eclipse.jetty.util.component.ContainerLifeCycle.start:169
  org.eclipse.jetty.util.component.ContainerLifeCycle.doStart:110
  org.eclipse.jetty.server.handler.AbstractHandler.doStart:97
  org.eclipse.jetty.util.component.AbstractLifeCycle.start:72
  org.eclipse.jetty.util.component.ContainerLifeCycle.start:169
  org.eclipse.jetty.util.component.ContainerLifeCycle.doStart:110
  org.eclipse.jetty.server.handler.AbstractHandler.doStart:97
  org.eclipse.jetty.server.handler.StatisticsHandler.doStart:255
  org.eclipse.jetty.util.component.AbstractLifeCycle.start:72
  org.eclipse.jetty.util.component.ContainerLifeCycle.start:169
  org.eclipse.jetty.server.Server.start:408
  org.eclipse.jetty.util.component.ContainerLifeCycle.doStart:110
  org.eclipse.jetty.server.handler.AbstractHandler.doStart:97
  org.eclipse.jetty.server.Server.doStart:372
  org.eclipse.jetty.util.component.AbstractLifeCycle.start:72
  io.dropwizard.cli.ServerCommand.run:53
  io.dropwizard.cli.EnvironmentCommand.run:45
  io.dropwizard.cli.ConfiguredCommand.run:87
  io.dropwizard.cli.Cli.run:78
  io.dropwizard.Application.run:94
  my.application.MainApplication.main:46
Caused by: java.lang.NoClassDefFoundError: org/glassfish/jersey/ext/cdi1x/internal/CdiUtil
  org.glassfish.jersey.server.validation.internal.hibernate.HibernateInjectingConstraintValidatorFactory.postConstruct:30
  jdk.internal.reflect.NativeMethodAccessorImpl.invoke0:native
  jdk.internal.reflect.NativeMethodAccessorImpl.invoke:62
  jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke:43
  java.lang.reflect.Method.invoke:566
  org.glassfish.hk2.utilities.reflection.ReflectionHelper.invoke:1268
  org.jvnet.hk2.internal.Utilities.justPostConstruct:892
  org.jvnet.hk2.internal.ServiceLocatorImpl.postConstruct:1026
  org.jvnet.hk2.internal.ServiceLocatorImpl.createAndInitialize:1074
  org.jvnet.hk2.internal.ServiceLocatorImpl.createAndInitialize:1064
  org.glassfish.jersey.inject.hk2.AbstractHk2InjectionManager.createAndInitialize:189
  org.glassfish.jersey.inject.hk2.ImmediateHk2InjectionManager.createAndInitialize:30
  org.glassfish.jersey.internal.inject.Injections.getOrCreate:106
  org.glassfish.jersey.server.JerseyResourceContextConfigurator.lambda$init$0:45
  org.glassfish.jersey.server.internal.JerseyResourceContext.getResource:83
  org.glassfish.jersey.server.validation.internal.CompositeInjectingConstraintValidatorFactory.postConstruct:45
  jdk.internal.reflect.NativeMethodAccessorImpl.invoke0:native
  jdk.internal.reflect.NativeMethodAccessorImpl.invoke:62
  jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke:43
  java.lang.reflect.Method.invoke:566
  org.glassfish.hk2.utilities.reflection.ReflectionHelper.invoke:1268
  org.jvnet.hk2.internal.Utilities.justPostConstruct:892
  org.jvnet.hk2.internal.ServiceLocatorImpl.postConstruct:1026
  org.jvnet.hk2.internal.ServiceLocatorImpl.createAndInitialize:1074
  org.jvnet.hk2.internal.ServiceLocatorImpl.createAndInitialize:1064
  org.glassfish.jersey.inject.hk2.AbstractHk2InjectionManager.createAndInitialize:189
  org.glassfish.jersey.inject.hk2.ImmediateHk2InjectionManager.createAndInitialize:30
  org.glassfish.jersey.internal.inject.Injections.getOrCreate:106
  org.glassfish.jersey.server.JerseyResourceContextConfigurator.lambda$init$0:45
  org.glassfish.jersey.server.internal.JerseyResourceContext.getResource:83
  org.glassfish.jersey.server.validation.internal.ValidationBinder$ConfiguredValidatorProvider.getDefaultValidatorContext:267
  org.glassfish.jersey.server.validation.internal.ValidationBinder$ConfiguredValidatorProvider.getDefaultValidator:245
  org.glassfish.jersey.server.validation.internal.ValidationBinder$ConfiguredValidatorProvider.get:187
  org.glassfish.jersey.server.validation.internal.ValidationBinder$ConfiguredValidatorProvider.get:157
  org.glassfish.jersey.inject.hk2.SupplierFactoryBridge.provide:76
  org.jvnet.hk2.internal.FactoryCreator.create:129
  org.jvnet.hk2.internal.SystemDescriptor.create:463
  org.jvnet.hk2.internal.PerLookupContext.findOrCreate:46
  org.jvnet.hk2.internal.Utilities.createService:2102
  org.jvnet.hk2.internal.ServiceLocatorImpl.internalGetService:758
  org.jvnet.hk2.internal.ServiceLocatorImpl.internalGetService:721
  org.jvnet.hk2.internal.ServiceLocatorImpl.getService:691
  org.glassfish.jersey.inject.hk2.AbstractHk2InjectionManager.getInstance:160
  org.glassfish.jersey.inject.hk2.ImmediateHk2InjectionManager.getInstance:30
  org.glassfish.jersey.server.model.internal.ResourceMethodInvokerConfigurator.lambda$postInit$0:54
  org.glassfish.jersey.server.model.ResourceMethodInvoker$Builder.build:194
  org.glassfish.jersey.server.internal.routing.RuntimeModelBuilder.createInflector:106
  org.glassfish.jersey.server.internal.routing.RuntimeModelBuilder.createMethodRouter:93
  org.glassfish.jersey.server.internal.routing.RuntimeModelBuilder.createResourceMethodRouters:287
  org.glassfish.jersey.server.internal.routing.RuntimeModelBuilder.buildModel:151
  org.glassfish.jersey.server.internal.routing.Routing$Builder.buildStage:223
  org.glassfish.jersey.server.ApplicationHandler.initialize:399
  org.glassfish.jersey.server.ApplicationHandler.lambda$initialize$1:293
  org.glassfish.jersey.internal.Errors.process:292
  org.glassfish.jersey.internal.Errors.process:274
  org.glassfish.jersey.internal.Errors.processWithException:232
  org.glassfish.jersey.server.ApplicationHandler.initialize:292
  org.glassfish.jersey.server.ApplicationHandler.<init>:259
  org.glassfish.jersey.servlet.WebComponent.<init>:311
  org.glassfish.jersey.servlet.ServletContainer.init:154
  org.glassfish.jersey.servlet.ServletContainer.init:347
  javax.servlet.GenericServlet.init:203
  org.eclipse.jetty.servlet.ServletHolder.initServlet:602
  org.eclipse.jetty.servlet.ServletHolder.initialize:413
  org.eclipse.jetty.servlet.ServletHandler.lambda$initialize$0:749
  java.util.stream.SortedOps$SizedRefSortingSink.end:357
  java.util.stream.AbstractPipeline.copyInto:485
  java.util.stream.AbstractPipeline.wrapAndCopyInto:474
  java.util.stream.StreamSpliterators$WrappingSpliterator.forEachRemaining:312
  java.util.stream.Streams$ConcatSpliterator.forEachRemaining:735
  java.util.stream.ReferencePipeline$Head.forEach:658
  org.eclipse.jetty.servlet.ServletHandler.initialize:773
  org.eclipse.jetty.servlet.ServletContextHandler.startContext:360
  org.eclipse.jetty.server.handler.ContextHandler.doStart:844
  org.eclipse.jetty.servlet.ServletContextHandler.doStart:275
  org.eclipse.jetty.util.component.AbstractLifeCycle.start:72
  org.eclipse.jetty.util.component.ContainerLifeCycle.start:169
  org.eclipse.jetty.util.component.ContainerLifeCycle.doStart:117
  org.eclipse.jetty.server.handler.AbstractHandler.doStart:97
  org.eclipse.jetty.util.component.AbstractLifeCycle.start:72
  org.eclipse.jetty.util.component.ContainerLifeCycle.start:169
  org.eclipse.jetty.util.component.ContainerLifeCycle.doStart:110
  org.eclipse.jetty.server.handler.AbstractHandler.doStart:97
  org.eclipse.jetty.server.handler.gzip.GzipHandler.doStart:425
  org.eclipse.jetty.util.component.AbstractLifeCycle.start:72
  org.eclipse.jetty.util.component.ContainerLifeCycle.start:169
  org.eclipse.jetty.util.component.ContainerLifeCycle.doStart:110
  org.eclipse.jetty.server.handler.AbstractHandler.doStart:97
  org.eclipse.jetty.util.component.AbstractLifeCycle.start:72
  org.eclipse.jetty.util.component.ContainerLifeCycle.start:169
  org.eclipse.jetty.util.component.ContainerLifeCycle.doStart:110
  org.eclipse.jetty.server.handler.AbstractHandler.doStart:97
  org.eclipse.jetty.server.handler.StatisticsHandler.doStart:255
  org.eclipse.jetty.util.component.AbstractLifeCycle.start:72
  org.eclipse.jetty.util.component.ContainerLifeCycle.start:169
  org.eclipse.jetty.server.Server.start:408
  org.eclipse.jetty.util.component.ContainerLifeCycle.doStart:110
  org.eclipse.jetty.server.handler.AbstractHandler.doStart:97
  org.eclipse.jetty.server.Server.doStart:372
  org.eclipse.jetty.util.component.AbstractLifeCycle.start:72
  io.dropwizard.cli.ServerCommand.run:53
  io.dropwizard.cli.EnvironmentCommand.run:45
  io.dropwizard.cli.ConfiguredCommand.run:87
  io.dropwizard.cli.Cli.run:78
  io.dropwizard.Application.run:94
  com.booking.attractions.payments.MainApplication.main:46
Caused by: java.lang.ClassNotFoundException: org.glassfish.jersey.ext.cdi1x.internal.CdiUtil
  jdk.internal.loader.BuiltinClassLoader.loadClass:581
  jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass:178
  java.lang.ClassLoader.loadClass:522
  org.glassfish.jersey.server.validation.internal.hibernate.HibernateInjectingConstraintValidatorFactory.postConstruct:30
  jdk.internal.reflect.NativeMethodAccessorImpl.invoke0:native
  jdk.internal.reflect.NativeMethodAccessorImpl.invoke:62
  jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke:43
  java.lang.reflect.Method.invoke:566
  org.glassfish.hk2.utilities.reflection.ReflectionHelper.invoke:1268
  org.jvnet.hk2.internal.Utilities.justPostConstruct:892
  org.jvnet.hk2.internal.ServiceLocatorImpl.postConstruct:1026
  org.jvnet.hk2.internal.ServiceLocatorImpl.createAndInitialize:1074
  org.jvnet.hk2.internal.ServiceLocatorImpl.createAndInitialize:1064
  org.glassfish.jersey.inject.hk2.AbstractHk2InjectionManager.createAndInitialize:189
  org.glassfish.jersey.inject.hk2.ImmediateHk2InjectionManager.createAndInitialize:30
  org.glassfish.jersey.internal.inject.Injections.getOrCreate:106
  org.glassfish.jersey.server.JerseyResourceContextConfigurator.lambda$init$0:45
  org.glassfish.jersey.server.internal.JerseyResourceContext.getResource:83
  org.glassfish.jersey.server.validation.internal.CompositeInjectingConstraintValidatorFactory.postConstruct:45
  jdk.internal.reflect.NativeMethodAccessorImpl.invoke0:native
  jdk.internal.reflect.NativeMethodAccessorImpl.invoke:62
  jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke:43
  java.lang.reflect.Method.invoke:566
  org.glassfish.hk2.utilities.reflection.ReflectionHelper.invoke:1268
  org.jvnet.hk2.internal.Utilities.justPostConstruct:892
  org.jvnet.hk2.internal.ServiceLocatorImpl.postConstruct:1026
  org.jvnet.hk2.internal.ServiceLocatorImpl.createAndInitialize:1074
  org.jvnet.hk2.internal.ServiceLocatorImpl.createAndInitialize:1064
  org.glassfish.jersey.inject.hk2.AbstractHk2InjectionManager.createAndInitialize:189
  org.glassfish.jersey.inject.hk2.ImmediateHk2InjectionManager.createAndInitialize:30
  org.glassfish.jersey.internal.inject.Injections.getOrCreate:106
  org.glassfish.jersey.server.JerseyResourceContextConfigurator.lambda$init$0:45
  org.glassfish.jersey.server.internal.JerseyResourceContext.getResource:83
  org.glassfish.jersey.server.validation.internal.ValidationBinder$ConfiguredValidatorProvider.getDefaultValidatorContext:267
  org.glassfish.jersey.server.validation.internal.ValidationBinder$ConfiguredValidatorProvider.getDefaultValidator:245
  org.glassfish.jersey.server.validation.internal.ValidationBinder$ConfiguredValidatorProvider.get:187
  org.glassfish.jersey.server.validation.internal.ValidationBinder$ConfiguredValidatorProvider.get:157
  org.glassfish.jersey.inject.hk2.SupplierFactoryBridge.provide:76
  org.jvnet.hk2.internal.FactoryCreator.create:129
  org.jvnet.hk2.internal.SystemDescriptor.create:463
  org.jvnet.hk2.internal.PerLookupContext.findOrCreate:46
  org.jvnet.hk2.internal.Utilities.createService:2102
  org.jvnet.hk2.internal.ServiceLocatorImpl.internalGetService:758
  org.jvnet.hk2.internal.ServiceLocatorImpl.internalGetService:721
  org.jvnet.hk2.internal.ServiceLocatorImpl.getService:691
  org.glassfish.jersey.inject.hk2.AbstractHk2InjectionManager.getInstance:160
  org.glassfish.jersey.inject.hk2.ImmediateHk2InjectionManager.getInstance:30
  org.glassfish.jersey.server.model.internal.ResourceMethodInvokerConfigurator.lambda$postInit$0:54
  org.glassfish.jersey.server.model.ResourceMethodInvoker$Builder.build:194
  org.glassfish.jersey.server.internal.routing.RuntimeModelBuilder.createInflector:106
  org.glassfish.jersey.server.internal.routing.RuntimeModelBuilder.createMethodRouter:93
  org.glassfish.jersey.server.internal.routing.RuntimeModelBuilder.createResourceMethodRouters:287
  org.glassfish.jersey.server.internal.routing.RuntimeModelBuilder.buildModel:151
  org.glassfish.jersey.server.internal.routing.Routing$Builder.buildStage:223
  org.glassfish.jersey.server.ApplicationHandler.initialize:399
  org.glassfish.jersey.server.ApplicationHandler.lambda$initialize$1:293
  org.glassfish.jersey.internal.Errors.process:292
  org.glassfish.jersey.internal.Errors.process:274
  org.glassfish.jersey.internal.Errors.processWithException:232
  org.glassfish.jersey.server.ApplicationHandler.initialize:292
  org.glassfish.jersey.server.ApplicationHandler.<init>:259
  org.glassfish.jersey.servlet.WebComponent.<init>:311
  org.glassfish.jersey.servlet.ServletContainer.init:154
  org.glassfish.jersey.servlet.ServletContainer.init:347
  javax.servlet.GenericServlet.init:203
  org.eclipse.jetty.servlet.ServletHolder.initServlet:602
  org.eclipse.jetty.servlet.ServletHolder.initialize:413
  org.eclipse.jetty.servlet.ServletHandler.lambda$initialize$0:749
  java.util.stream.SortedOps$SizedRefSortingSink.end:357
  java.util.stream.AbstractPipeline.copyInto:485
  java.util.stream.AbstractPipeline.wrapAndCopyInto:474
  java.util.stream.StreamSpliterators$WrappingSpliterator.forEachRemaining:312
  java.util.stream.Streams$ConcatSpliterator.forEachRemaining:735
  java.util.stream.ReferencePipeline$Head.forEach:658
  org.eclipse.jetty.servlet.ServletHandler.initialize:773
  org.eclipse.jetty.servlet.ServletContextHandler.startContext:360
  org.eclipse.jetty.server.handler.ContextHandler.doStart:844
  org.eclipse.jetty.servlet.ServletContextHandler.doStart:275
  org.eclipse.jetty.util.component.AbstractLifeCycle.start:72
  org.eclipse.jetty.util.component.ContainerLifeCycle.start:169
  org.eclipse.jetty.util.component.ContainerLifeCycle.doStart:117
  org.eclipse.jetty.server.handler.AbstractHandler.doStart:97
  org.eclipse.jetty.util.component.AbstractLifeCycle.start:72
  org.eclipse.jetty.util.component.ContainerLifeCycle.start
```

In our current setup we are exclusively using HK2, therefore having CDI loaded would be only harmful to us.

--- a/ext/bean-validation/src/main/java/org/glassfish/jersey/server/validation/internal/ValidationBinder.java
+++ b/ext/bean-validation/src/main/java/org/glassfish/jersey/server/validation/internal/ValidationBinder.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2012, 2018 Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2012, 2020 Oracle and/or its affiliates. All rights reserved.
  * Copyright (c) 2018, 2019 Payara Foundation and/or its affiliates. All rights reserved.
  *
  * This program and the accompanying materials are made available under the
@@ -18,7 +18,6 @@
 package org.glassfish.jersey.server.validation.internal;
 
 import java.security.AccessController;
-import java.security.PrivilegedAction;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.WeakHashMap;
@@ -259,9 +258,8 @@ public final class ValidationBinder extends AbstractBinder {
         private ValidatorContext getDefaultValidatorContext(final ValidateOnExecutionHandler handler) {
             final ValidatorContext context = factory.usingContext();
 
-            // if CDI is available use composite factiry
-            if (AccessController.doPrivileged(
-                    ReflectionHelper.classForNamePA("javax.enterprise.inject.spi.BeanManager")) != null) {
+            // if CDI is available use composite factory
+            if (isCDIAvailable()) {
                 // Composite Configuration - due to PAYARA-2491
                 // https://github.com/payara/Payara/issues/2245
                 context.constraintValidatorFactory(resourceContext.getResource(
@@ -275,6 +273,15 @@ public final class ValidationBinder extends AbstractBinder {
             context.traversableResolver(getTraversableResolver(factory.getTraversableResolver(), handler));
 
             return context;
+        }
+
+        private boolean isCDIAvailable() {
+            // Both CDI & Jersey CDI modules must be available
+            return AccessController.doPrivileged(
+                        ReflectionHelper.classForNamePA("javax.enterprise.inject.spi.BeanManager")) != null
+                   &&
+                   AccessController.doPrivileged(
+                        ReflectionHelper.classForNamePA("org.glassfish.jersey.ext.cdi1x.internal.CdiUtil")) != null;
         }
 
         /**

--- a/tests/integration/jersey-4542/pom.xml
+++ b/tests/integration/jersey-4542/pom.xml
@@ -1,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+
+    This program and the accompanying materials are made available under the
+    terms of the Eclipse Public License v. 2.0, which is available at
+    http://www.eclipse.org/legal/epl-2.0.
+
+    This Source Code may also be made available under the following Secondary
+    Licenses when the conditions for such availability set forth in the
+    Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+    version 2 with the GNU Classpath Exception, which is available at
+    https://www.gnu.org/software/classpath/license.html.
+
+    SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>project</artifactId>
+        <groupId>org.glassfish.jersey.tests.integration</groupId>
+        <version>2.32.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>jersey-4542</artifactId>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.glassfish.jersey.ext</groupId>
+            <artifactId>jersey-bean-validation</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.glassfish.jersey.test-framework.providers</groupId>
+            <artifactId>jersey-test-framework-provider-grizzly2</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.glassfish.jersey.test-framework.providers</groupId>
+            <artifactId>jersey-test-framework-provider-external</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.glassfish.jersey.test-framework</groupId>
+            <artifactId>jersey-test-framework-core</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.weld.se</groupId>
+            <artifactId>weld-se-core</artifactId>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+</project>

--- a/tests/integration/jersey-4542/src/main/java/org.glassfish.jersey.tests.integration.jersey4542/ValidationInflector.java
+++ b/tests/integration/jersey-4542/src/main/java/org.glassfish.jersey.tests.integration.jersey4542/ValidationInflector.java
@@ -1,0 +1,51 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+
+package org.glassfish.jersey.tests.integration.jersey4542;
+
+import java.io.IOException;
+
+import javax.ws.rs.container.ContainerRequestContext;
+
+import javax.validation.constraints.NotNull;
+
+import org.glassfish.jersey.message.internal.ReaderWriter;
+import org.glassfish.jersey.process.Inflector;
+
+/**
+ * @author Michal Gajdos
+ */
+public class ValidationInflector implements Inflector<ContainerRequestContext, String> {
+
+    @NotNull
+    @Override
+    public String apply(final ContainerRequestContext requestContext) {
+        return get(requestContext);
+    }
+
+    @NotNull
+    public String get(@NotNull final ContainerRequestContext requestContext) {
+        try {
+            final String entity = ReaderWriter.readFromAsString(
+                    requestContext.getEntityStream(),
+                    requestContext.getMediaType());
+
+            return entity.isEmpty() ? null : entity;
+        } catch (IOException e) {
+            return "error";
+        }
+    }
+}

--- a/tests/integration/jersey-4542/src/test/java/org.glassfish.jersey.tests.integration.jersey4542/ProgrammaticValidationTest.java
+++ b/tests/integration/jersey-4542/src/test/java/org.glassfish.jersey.tests.integration.jersey4542/ProgrammaticValidationTest.java
@@ -1,0 +1,103 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+
+package org.glassfish.jersey.tests.integration.jersey4542;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.Application;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.glassfish.jersey.inject.hk2.Hk2InjectionManagerFactory;
+import org.glassfish.jersey.logging.LoggingFeature;
+import org.glassfish.jersey.server.ResourceConfig;
+import org.glassfish.jersey.server.model.Resource;
+import org.glassfish.jersey.test.JerseyTest;
+
+import static org.junit.Assert.assertEquals;
+
+import org.glassfish.jersey.test.external.ExternalTestContainerFactory;
+import org.jboss.weld.environment.se.Weld;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Bean Validation tests for programmatically created resources.
+ *
+ * @author Michal Gajdos
+ */
+public class ProgrammaticValidationTest extends JerseyTest {
+
+    Weld weld;
+
+    @Before
+    public void setup() {
+        Assume.assumeTrue(Hk2InjectionManagerFactory.isImmediateStrategy());
+    }
+
+    @Override
+    public void setUp() throws Exception {
+        if (Hk2InjectionManagerFactory.isImmediateStrategy()) {
+            if (!ExternalTestContainerFactory.class.isAssignableFrom(getTestContainerFactory().getClass())) {
+                weld = new Weld();
+                weld.initialize();
+            }
+            super.setUp();
+        }
+    }
+
+    @Override
+    public void tearDown() throws Exception {
+        if (Hk2InjectionManagerFactory.isImmediateStrategy()) {
+            if (!ExternalTestContainerFactory.class.isAssignableFrom(getTestContainerFactory().getClass())) {
+                weld.shutdown();
+            }
+            super.tearDown();
+        }
+    }
+
+    @Override
+    protected Application configure() {
+        final Set<Resource> resources = new HashSet<>();
+
+        Resource.Builder resourceBuilder = Resource.builder("class");
+        resourceBuilder
+                .addMethod("POST")
+                .handledBy(ValidationInflector.class);
+        resources.add(resourceBuilder.build());
+
+        return new ResourceConfig().register(LoggingFeature.class).registerResources(resources);
+    }
+
+    @Test
+    public void testInflectorClass() throws Exception {
+        final Response response = target("class").request().post(Entity.entity("value", MediaType.TEXT_PLAIN_TYPE));
+
+        assertEquals(200, response.getStatus());
+        assertEquals("value", response.readEntity(String.class));
+    }
+
+    @Test
+    public void testInflectorClassNegative() throws Exception {
+        final Response response = target("class").request().post(Entity.entity(null, MediaType.TEXT_PLAIN_TYPE));
+
+        assertEquals(500, response.getStatus());
+    }
+}

--- a/tests/integration/jersey-4542/src/test/resources/META-INF/beans.xml
+++ b/tests/integration/jersey-4542/src/test/resources/META-INF/beans.xml
@@ -1,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+
+    This program and the accompanying materials are made available under the
+    terms of the Eclipse Public License v. 2.0, which is available at
+    http://www.eclipse.org/legal/epl-2.0.
+
+    This Source Code may also be made available under the following Secondary
+    Licenses when the conditions for such availability set forth in the
+    Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+    version 2 with the GNU Classpath Exception, which is available at
+    https://www.gnu.org/software/classpath/license.html.
+
+    SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+
+-->
+
+<beans/>

--- a/tests/integration/pom.xml
+++ b/tests/integration/pom.xml
@@ -86,6 +86,7 @@
         <module>jersey-4099</module>
         <module>jersey-4321</module>
         <module>jersey-4507</module>
+        <module>jersey-4542</module>
         <module>jetty-response-close</module>
         <module>microprofile</module>
         <module>portability-jersey-1</module>

